File: app/models/ticket.rb
class Ticket < ApplicationRecord
    validates :title, presence: true
    validates :body, presence: true
    belongs_to :ticket_type
    has_many :posts

    def as_json(include_options = {}, other_options = {})
      include_options = include_options.merge(
        ticket_type: { only: :name }
      )

      super(
        include: include_options
      )
    end
end

File: react-frontend/src/types/Ticket.ts
import type { Post } from "../types/Post.ts"

export interface TicketType {
  name: string
}

export interface Ticket {
  id: number,
  title: string,
  body: string,
  created_at: string,
  ticket_type: TicketType,
  updated_at?: string,
  posts?: Array<Post>,
}
